src/lib.rs

```diff
@@ -403,6 +403,13 @@ where
         self.get(key, guard).map(then)
     }
 
+    /// Returns the key-value pair corresponding to the supplied key.
+    ///
+    /// Returns `None` if this map contains no mapping for `key`.
```

Comment:
It'd be good to match the documentation and bounds on [`HashMap::get_key_value`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.get_key_value) here.